indico/modules/events/models/persons.py

```diff
@@ -146,6 +147,69 @@ def for_user(cls, user, event):
         person = event.persons.filter_by(user=user).first()
         return person or cls.create_from_user(user, event)
 
+    @classmethod
+    def replace_email_with_user(cls, user):
+        """
+        Replaces all email-based persons matching the user's email
```

Comment:
that sentence makes no sense and on a second thought neither does the method name (we don't *replace* emails). can't think of a better name atm though. `set_user_from_email` sounds a bit weird too